homeassistant/components/ps4/__init__.py

```diff
@@ -54,6 +61,40 @@
                 _LOGGER.info(
                     "PlayStation 4 Config Updated: \
                     Region changed to: %s", country)
+                version = 2
+
+    # Migrate Version 2 -> Version 3: Update identifier format.
+    if version == 2:
+        # Prevent changing entity_id. Updates entity registry.
+        registry = await entity_registry.async_get_registry(hass)
+
+        for entity_id, e_entry in registry.entities.items():
+            if e_entry.config_entry_id == entry.entry_id:
+                unique_id = e_entry.unique_id
+                entity_id = registry.async_get_entity_id(
+                    'media_player', DOMAIN, unique_id)
+
+                # Remove old entity entry.
+                registry.async_remove(entity_id)
+                await hass.async_block_till_done()
+
+                # Use last 4 of credential for suffix.
+                suffix = entry.data[CONF_TOKEN][-4:]
+                unique_id = '{}_{}'.format(unique_id, suffix)
```

Comment:
Make a helper function for creating the new `unique_id` so we don't have that logic in two places. Have it accept the mac address and the token and return the unique_id.